packages/as-types/assembly/safeMath.ts

```diff
@@ -71,9 +72,9 @@ export class SafeMath {
     if (a == 0) {
       return 0;
     }
+    assert(b < u64.MAX_VALUE / a, 'SafeMath: multiplication overflow');
```

Comment:
You start with:
a*b <= u.MAX
<=> b <= u.max/a, a > 0

```suggestion
    assert(b <= u64.MAX_VALUE / a, 'SafeMath: multiplication overflow');
```